### ISSUE TITLE
Running istio integration/conformance tests

### ISSUE THREAD

**Issue by johscheuer** [istio/istio#15815]
**Bug description**

I wanted to run the `conformance` tests in the integration tests (https://github.com/istio/istio/tree/master/tests/integration) and some of the tests are failing. After digging into the code I saw that currently galley will be queried with "insecure" gRPC but our galley instance uses mTLS.

Here are some galley logs:

```bash
2019-07-24T10:52:27.945531Z info
grpc: Server.Serve failed to complete security handshake from "127.0.0.1:35644": tls: first record does not look like a TLS handshake
2019-07-24T10:52:31.101903Z info
grpc: Server.Serve failed to complete security handshake from "127.0.0.1:35698": tls: first record does not look like a TLS handshake
```

This function will query galley: https://github.com/istio/istio/blob/master/tests/integration/conformance/conformance_test.go#L149 

and in this function a new client with a insecure gRPC connection is instantiated: https://github.com/istio/istio/blob/master/pkg/test/framework/components/galley/client.go#L106

** Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
I would expect that the test Framework validates if mTLS (or TLS in general) is activated and if so the test should be skipped or even better the certificates should be fetched out of Kubernetes and be used to make a secure connection to galley (if possible).

**Steps to reproduce the bug**
Create an istio cluster with a secured galley and run the conformance tests e.g.:

```bash
go test ./test/integration/conformance --istio.test..env kube --istio.test.kube.deploy=false
```

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

Istio 1.1.7
Kubernetes 1.14.3

**How was Istio installed?**

With helm charts (and some patching of the yaml resources)

**Environment where bug was observed (cloud vendor, OS, etc)**
- bare-metal
- ContainerLinux

**Comment by johscheuer** [comment]
I would start working on this with a simple solution where you need to provide the according key and certificate

**Comment by johscheuer** [comment]
I close this since there are known "workarounds'

**Comment by sushicw** [comment]
@johscheuer For the benefit of anyone who stumbles across this in the future, could you please link to the workarounds you're referring to? Or, if there's nothing to link to, at least summarize them. Thanks!

**Comment by johscheuer** [comment]
The solutions are here in the closed PR: https://github.com/istio/istio/pull/16236#discussion_r314420235 
TLDR if you use the new installer (istioctl) you don't need to adjust anything
